server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -656,6 +674,15 @@ class DeltaSharedTableKernel(
     val addFileVectorPath = addFileColumnVectors.path.getString(rowId)
     val cloudPath = absolutePath(dataPath, addFileVectorPath)
     val signedUrl = fileSigner.sign(cloudPath)
+    var dvDescriptor = getDeltaDeletionVectorDescriptor(addFileColumnVectors.deletionVector, rowId)
+
+    if (dvDescriptor != null) {
```

Comment:
We should double check `clientReaderFeatures` here, because table can possibly contain DV even when DV properties are disabled. This happens when the user explicitly sets DV property to false after a DV is generated.

Reply:
fixed!